src/main/java/filter/expression/QualifierContent.java

```diff
@@ -0,0 +1,66 @@
+package filter.expression;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class QualifierContent {
+
+    //Only one of these will exist in a QualifierContent object
```

Comment:
[Inheritance](http://programmers.stackexchange.com/a/163987) should be nicer here, since the values are parameterised, and each type of content should know how to create its corresponding qualifier (good use case for method dispatch).

Anyway, this (anti-)pattern is repeated inside Qualifier as well, but there it's a bit harder to get rid of without resorting to instanceof.